### ISSUE TITLE
Attempting to load directory with loadfile() causes out-of-memory abort.

### ISSUE THREAD

**Issue by shacknetisp** [red-eclipse/base#616]
This stems from when `seek(0, SEEK_END)` is called upon a directory, `tell()` will return an enormous number, which is then returned as the stream's size.
`shared/stream.cpp:1259` is `char *buf = new char[len+1];` where `len` is the result of `f->size()`, causing the allocation to abort.

**Comment by shacknetisp** [comment]
This issue is still occurring even with this commit, it is easy to reproduce by trying to `exec ""` or `exec data`.

An example of the root cause:
```
FILE *f = fopen("/tmp", "rb");
printf("%d\n", fseek(f, 0, SEEK_END)); // Prints '0'
printf("%ld\n", ftell(f)); // Prints '9223372036854775807'
fclose(f);
```

From what I can see there will need to be some detection if it is trying to load a directory, or a new method of file size determination altogether.

**Comment by qreeves** [comment]
This issue isn't present in Windows, so I'm not fully able to test my fix in 0b6e1f0 - so let me know and close if fixed.

**Comment by lsalzman** [comment]
Fixed.
